src/app-gocardless/services/gocardless-service.js

```diff
@@ -211,6 +212,20 @@ export const goCardlessService = {
       goCardlessService.getBalances(accountId),
     ]);
 
+    transactions.transactions.booked = transactions.transactions?.booked?.map(
```

Comment:
Won't this crash?

If `transactions.transactions` is undefined, the right hand side will eventually return an undefined value.  But you then try to set that value to `transactions.transactions.booked` which should give an undefined error when accessing `transactions.transactions`.

I think you need to guard this block with `if (transactions.transactions)`.

Reply:
You're right, I was copying the sort logic from below but looking into the `getTransactions` function, it doesn't look like this could ever be undefined.

https://github.com/actualbudget/actual-server/blob/db6b4e4800448625fe0be969938b7673c3badbae/src/app-gocardless/services/gocardless-service.js#L458-L465

Would you prefer i guard this or remove the optional chaining operator entirely?

Reply:
I don't know that API but if it is never undefined, then dropping the `?`'s makes sense.

Does it always return an object with `booked` and `pending` set, even if there are no transactions of those kind?

Reply:
Just checked, booked and pending will always be an array and will be left empty if there are no transactions so the chaining here isn't required. Removed it now